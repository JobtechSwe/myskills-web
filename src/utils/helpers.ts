import Cookies from 'js-cookie'
import { LocalStateProps, ContactInformationProps } from '../graphql/client'
import {
  EducationInput,
  Occupation,
  OntologyRelationResponse,
} from '../generated/myskills'

export const getCookie = (name: string) => Cookies.get(name)

export const setCookie = (
  name: string,
  value: string,
  options: object = {}
) => {
  Cookies.set(name, value, options)
}

export const removeCookie = (name: string) => {
  Cookies.remove(name)
}

export const redirect = (route: string) => (location.href = route)

interface StorageEntryProps {
  type: string
}

interface Educations extends StorageEntryProps {
  type: 'educations'
  data: EducationInput[]
}

interface OccupationStorage extends StorageEntryProps {
  type: 'occupation'
  data: Occupation
}

interface Skills extends StorageEntryProps {
  type: 'skills'
  data: OntologyRelationResponse[]
}
interface Traits extends StorageEntryProps {
  type: 'traits'
  data: string[]
}
interface WhoAmI extends StorageEntryProps {
  type: 'whoAmI'
  data: string
}

interface ContactInformation extends StorageEntryProps {
  type: 'contact'
  data: ContactInformationProps
}

export type StorageEntry =
  | ContactInformation
  | Educations
  | OccupationStorage
  | Skills
  | Traits
  | WhoAmI

export const storageHelper = {
  load: (initialState: LocalStateProps) =>
    Object.entries(initialState).reduce(
      (acc, [prop, val]) => ({
        ...acc,
        [prop]: JSON.parse(localStorage.getItem(prop) as string) || val,
      }),
      {} as LocalStateProps
    ),
  set: (payload: StorageEntry) =>
    localStorage.setItem(payload.type, JSON.stringify(payload.data)),
}
