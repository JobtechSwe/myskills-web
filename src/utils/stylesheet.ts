const colors = {
  lassekongo: 'dodgerblue',
  primary: 'pink',
}

const sizes = ['0', '1rem', '2rem', '4rem']

const typography = {
  default: 'Arial',
}

const space = {
  none: sizes[0],
  small: sizes[1],
  medium: sizes[2],
  big: sizes[3],
}

export default { colors, space, typography }
