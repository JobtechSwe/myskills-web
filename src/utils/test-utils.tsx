import * as React from 'react'
import ApolloClient from 'apollo-client'
import { ApolloProvider } from 'react-apollo-hooks'
import { InMemoryCache } from 'apollo-cache-inmemory'
import { MockLink, MockedProvider } from 'react-apollo/test-utils'
import { ThemeProvider } from 'emotion-theming'
import { render as rtlRender } from 'react-testing-library'
import { theme } from 'theme'

export const render = (children: React.ReactNode, mocks?: any[]) => {
  const client = new ApolloClient({
    cache: new InMemoryCache({ addTypename: true }),
    link: new MockLink(mocks || []),
  })

  return rtlRender(
    <MockedProvider addTypename={true} mocks={mocks}>
      <ApolloProvider client={client}>
        <ThemeProvider theme={theme}>{children}</ThemeProvider>
      </ApolloProvider>
    </MockedProvider>
  )
}
