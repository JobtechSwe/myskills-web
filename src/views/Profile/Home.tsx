import React from 'react'
import { RouteComponentProps } from '@reach/router'
import ProfileLayout from 'components/Layout/ProfileLayout'
import gql from 'graphql-tag'
import { useQuery } from 'react-apollo-hooks'
import styled from '@emotion/styled'
import Flex from 'components/Flex'
import ProfileDataCard from 'components/ProfileDataCard'
import { Paragraph, H2 } from 'components/Typography'
import { Skill, Education, Experience } from 'generated/myskills'
import Loader from 'components/Loader'
import editIcon from 'assets/icons/edit.svg'
import { InternalLink } from 'components/Link'

export const GET_CV = gql`
  query getCV {
    occupation {
      term
      experience {
        years
      }
    }

    experiences {
      id
      employer
      sourceId
      term
      start
      end
    }
    image

    educations {
      programme
      school
      start
      end
      id
    }

    skills {
      sourceId
      term
      type
      id
    }

    profile {
      name
      email
      telephone
    }

    traits
    personalDescription
  }
`

const Container = styled(Flex)`
  div:nth-of-type(odd) {
    margin-right: 10px;
  }
`

const Image = styled.img`
  width: 100%;
  height: 100%;
`

const Home: React.FC<RouteComponentProps> = ({
  location = { pathname: '' },
}) => {
  const { data, error, loading } = useQuery(GET_CV, {
    fetchPolicy: 'network-only',
  })

  return (
    <ProfileLayout currentPath={location.pathname}>
      {error && 'error'}
      {loading && <Loader />}
      {!loading && data && (
        <>
          <Container mb="10px">
            <ProfileDataCard
              flexVal={0.65}
              isTop
              noBackground
              route="beskriv-dig"
            >
              <Paragraph fontSize="small" mb="none" mt="none">
                {data.personalDescription}
              </Paragraph>
            </ProfileDataCard>
            <ProfileDataCard flexVal={0.35} isImage isTop route="editplace">
              <Image src={`data:image/jpeg;base64,${data.image}`} />
            </ProfileDataCard>
          </Container>
          <Container mb="10px">
            <ProfileDataCard route="egenskaper">
              <H2 mb="small">Egenskaper</H2>
              {data.traits &&
                data.traits.map((trait: string) => (
                  <Paragraph fontSize="small" key={trait} mb="none" mt="none">
                    {trait},
                  </Paragraph>
                ))}
            </ProfileDataCard>
            <ProfileDataCard route="editplace">
              <H2 mb="small">Kompetens</H2>
              {data.skills &&
                data.skills.map((skill: Skill) => (
                  <Paragraph
                    fontSize="small"
                    key={skill.id}
                    mb="none"
                    mt="none"
                  >
                    {skill.term},
                  </Paragraph>
                ))}
            </ProfileDataCard>
          </Container>
          <Container mb="10px">
            <ProfileDataCard route="utbildning">
              <H2 mb="small">Utbildning</H2>
              {data.educations &&
                data.educations.map((education: Education) => (
                  <Paragraph
                    fontSize="small"
                    key={education.id}
                    mb="none"
                    mt="none"
                  >
                    {education.school}
                  </Paragraph>
                ))}
            </ProfileDataCard>
            <ProfileDataCard route="erfarenheter">
              <H2 mb="small">Erfarenhet</H2>
              {data.occupation && (
                <Paragraph mb="small" mt="none">
                  {`${
                    data.occupation.experience
                      ? data.occupation.experience.years
                      : 0
                  } års erfarenhet som
                  ${data.occupation.term}.`}
                </Paragraph>
              )}

              {data.experiences &&
                data.experiences.map((experience: Experience) => (
                  <Paragraph
                    fontSize="small"
                    key={experience.id}
                    mb="none"
                    mt="none"
                  >
                    {experience.term}
                  </Paragraph>
                ))}
            </ProfileDataCard>
          </Container>
          <Flex mt="small" pl="small">
            <Flex flex="1" flexDirection="column" mb="large">
              <H2 mb="none" width="100%">
                Kontakt
              </H2>
              {data.profile && (
                <Flex
                  alignItems="flex-end"
                  flexWrap="wrap"
                  justifyContent="space-between"
                >
                  <Paragraph fontSize="small" mb="none" mr="5px" mt="8px">
                    {data.profile.email}
                  </Paragraph>
                  <Paragraph fontSize={25} mb="none" mr="5px" mt="8px">
                    ·
                  </Paragraph>
                  <Paragraph fontSize="small" mb="none" mr="5px" mt="8px">
                    {data.profile.telephone}
                  </Paragraph>

                  <InternalLink mb="4px" to="editplace">
                    <img alt="Edit profile" src={editIcon} />
                  </InternalLink>
                </Flex>
              )}
            </Flex>
          </Flex>
        </>
      )}
    </ProfileLayout>
  )
}

export default Home
