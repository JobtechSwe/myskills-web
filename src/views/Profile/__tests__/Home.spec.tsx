import * as React from 'react'
import { render } from '../../../utils/test-utils'
import { wait } from 'react-testing-library'
import Home, { GET_PROFILE } from '../Home'
import mockData from '../__fixtures__/profileData.js'

describe('views/profile/Home', () => {
  let withResultsMock: any

  beforeEach(() => {
    withResultsMock = [
      {
        request: {
          query: GET_PROFILE,
        },
        result: {
          data: {
            ...mockData,
          },
        },
      },
    ]
  })

  it('should render with profile data', async () => {
    const { getByText, container } = render(<Home />, withResultsMock)
    await wait()
    expect(getByText('Webbutvecklare')).toBeInTheDocument()
    expect(container).toMatchSnapshot()
  })
})
