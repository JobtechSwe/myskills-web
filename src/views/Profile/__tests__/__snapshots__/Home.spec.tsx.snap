// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`views/profile/Home should render with profile data 1`] = `
<div>
  <div
    class="css-1wk798u"
  >
    <div
      class="css-1bqxkm"
    >
      <button
        class="css-ffgl9m"
      >
        <img
          alt="save_as_pdf"
          src="save_pdf.svg"
        />
      </button>
      <div
        class="css-1jo84sd"
      >
        <p
          class="css-twz9yi"
          font-family="default"
          font-size="medium"
        >
          Inget yrke angivet
        </p>
        <img
          alt="edit"
          class="css-ns6bdc"
          src="edit.svg"
        />
      </div>
      <p
        class="css-qlepwc"
        font-family="default"
        font-size="large"
        font-weight="bold"
      >
        Inget namn angivet
      </p>
    </div>
    <div
      class="css-29r7oe"
    >
      <div
        class="css-46fyds"
      >
        <a
          class="css-jbzzha"
          color="persianBlue"
          font-weight="500"
          href="/editplace"
        >
          <img
            alt="Edit data"
            src="edit.svg"
          />
        </a>
        <p
          class="css-wacz66"
          font-family="default"
          font-size="small"
        >
          Jag är en snäll och gullig person som gillar att äta bröd och godis och frukost och middag. Jag är ansvarstagande och stresshantering 
        </p>
      </div>
      <div
        class="css-metx23"
      >
        <a
          class="css-jbzzha"
          color="persianBlue"
          font-weight="500"
          href="/editplace"
        >
          <img
            alt="Edit data"
            src="edit_image.svg"
          />
        </a>
        Bild
      </div>
    </div>
    <div
      class="css-29r7oe"
    >
      <div
        class="css-mefo75"
      >
        <a
          class="css-jbzzha"
          color="persianBlue"
          font-weight="500"
          href="/editplace"
        >
          <img
            alt="Edit data"
            src="edit.svg"
          />
        </a>
        <h2
          class="css-an7vzf"
          color="black"
          font-family="default"
          font-size="medium"
        >
          Egenskaper
        </h2>
        <p
          class="css-wacz66"
          font-family="default"
          font-size="small"
        >
          Ansvarstagande
          ,
        </p>
        <p
          class="css-wacz66"
          font-family="default"
          font-size="small"
        >
          Frukost
          ,
        </p>
        <p
          class="css-wacz66"
          font-family="default"
          font-size="small"
        >
          Stresshantering
          ,
        </p>
        <p
          class="css-wacz66"
          font-family="default"
          font-size="small"
        >
          Snäll
          ,
        </p>
        <p
          class="css-wacz66"
          font-family="default"
          font-size="small"
        >
          Godis
          ,
        </p>
      </div>
      <div
        class="css-mefo75"
      >
        <a
          class="css-jbzzha"
          color="persianBlue"
          font-weight="500"
          href="/editplace"
        >
          <img
            alt="Edit data"
            src="edit.svg"
          />
        </a>
        <h2
          class="css-an7vzf"
          color="black"
          font-family="default"
          font-size="medium"
        >
          Kompetens
        </h2>
        <p
          class="css-wacz66"
          font-family="default"
          font-size="small"
        >
          Microsoft azure
          ,
        </p>
        <p
          class="css-wacz66"
          font-family="default"
          font-size="small"
        >
          Microsoft biztalk
          ,
        </p>
        <p
          class="css-wacz66"
          font-family="default"
          font-size="small"
        >
          Integrationslösningar
          ,
        </p>
        <p
          class="css-wacz66"
          font-family="default"
          font-size="small"
        >
          Integrationsverktyg
          ,
        </p>
      </div>
    </div>
    <div
      class="css-29r7oe"
    >
      <div
        class="css-mefo75"
      >
        <a
          class="css-jbzzha"
          color="persianBlue"
          font-weight="500"
          href="/editplace"
        >
          <img
            alt="Edit data"
            src="edit.svg"
          />
        </a>
        <h2
          class="css-an7vzf"
          color="black"
          font-family="default"
          font-size="medium"
        >
          Utbildning
        </h2>
        <p
          class="css-wacz66"
          font-family="default"
          font-size="small"
        >
          Webbutvecklare
        </p>
      </div>
      <div
        class="css-mefo75"
      >
        <a
          class="css-jbzzha"
          color="persianBlue"
          font-weight="500"
          href="/editplace"
        >
          <img
            alt="Edit data"
            src="edit.svg"
          />
        </a>
        <p
          class="css-63olqc"
          font-family="default"
          font-size="small"
        >
          4 års erfarenhet som 
                  Systemutvecklare.
        </p>
        <p
          class="css-wacz66"
          font-family="default"
          font-size="small"
        >
          Kung
        </p>
      </div>
    </div>
    <div
      class="css-1iewklz"
    >
      <a
        class="css-tzjau4"
        color="persianBlue"
        font-weight="500"
        href="/profil/"
        width="100%"
      >
        <button
          class="css-1ifp9gy"
          font-family="default"
          font-size="medium"
          width="inherit"
        >
          <div
            class="css-6591i1"
          >
            <svg
              fill="none"
              height="18"
              viewBox="0 0 18 18"
              width="18"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M9 9C10.6575 9 12 7.6575 12 6C12 4.3425 10.6575 3 9 3C7.3425 3 6 4.3425 6 6C6 7.6575 7.3425 9 9 9ZM9 10.5C6.9975 10.5 3 11.505 3 13.5V15H15V13.5C15 11.505 11.0025 10.5 9 10.5Z"
                fill="black"
              />
            </svg>
            Min profil
          </div>
        </button>
      </a>
      <a
        class="css-tzjau4"
        color="persianBlue"
        font-weight="500"
        href="//tidslinje"
        width="100%"
      >
        <button
          class="css-1ifp9gy"
          font-family="default"
          font-size="medium"
          width="inherit"
        >
          <div
            class="css-6591i1"
          >
            <svg
              fill="none"
              height="10"
              viewBox="0 0 21 10"
              width="21"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M20.0571 1.66667C20.0571 2.58333 19.2723 3.33333 18.313 3.33333C18.156 3.33333 18.0077 3.31667 17.8682 3.275L14.7635 6.23333C14.8071 6.36667 14.8246 6.51667 14.8246 6.66667C14.8246 7.58333 14.0397 8.33333 13.0804 8.33333C12.1211 8.33333 11.3362 7.58333 11.3362 6.66667C11.3362 6.51667 11.3536 6.36667 11.3973 6.23333L9.17342 4.10833C9.03388 4.15 8.87691 4.16667 8.71993 4.16667C8.56295 4.16667 8.40598 4.15 8.26644 4.10833L4.29842 7.90833C4.34202 8.04167 4.35947 8.18333 4.35947 8.33333C4.35947 9.25 3.57458 10 2.61528 10C1.65598 10 0.871094 9.25 0.871094 8.33333C0.871094 7.41667 1.65598 6.66667 2.61528 6.66667C2.77226 6.66667 2.92051 6.68333 3.06005 6.725L7.03679 2.93333C6.99319 2.8 6.97574 2.65 6.97574 2.5C6.97574 1.58333 7.76063 0.833333 8.71993 0.833333C9.67923 0.833333 10.4641 1.58333 10.4641 2.5C10.4641 2.65 10.4467 2.8 10.4031 2.93333L12.6269 5.05833C12.7664 5.01667 12.9234 5 13.0804 5C13.2374 5 13.3943 5.01667 13.5339 5.05833L16.6298 2.09167C16.5862 1.95833 16.5688 1.81667 16.5688 1.66667C16.5688 0.75 17.3536 0 18.313 0C19.2723 0 20.0571 0.75 20.0571 1.66667Z"
                fill="black"
              />
            </svg>
            Tidslinje
          </div>
        </button>
      </a>
    </div>
  </div>
</div>
`;
